File: app/src/main/java/com/mirash/familiar/activity/edit/CredentialsItemCallback.java
package com.mirash.familiar.activity.edit;

import com.mirash.familiar.motion.OnStartDragListener;
import com.mirash.familiar.model.CredentialsItem;

import java.util.List;

/**
 * @author Mirash
 */
public interface CredentialsItemCallback extends OnStartDragListener {
    void onLinkClick(String link);

    void onEditClick(CredentialsItem item);

    void onOrderChanged(List<CredentialsItem> items);

    void onShare(CredentialsItem item);
}

File: app/src/main/java/com/mirash/familiar/activity/main/adapter/CredentialsItemCallback.kt
package com.mirash.familiar.activity.main.adapter

import com.mirash.familiar.model.CredentialsItem
import com.mirash.familiar.motion.OnStartDragListener

/**
 * @author Mirash
 */
interface CredentialsItemCallback : OnStartDragListener {
    fun onLinkClick(link: String)
    fun onEditClick(item: CredentialsItem)
    fun onOrderChanged(items: List<CredentialsItem>)
    fun onShare(item: CredentialsItem)
}
